events/include/events/Event.h

```diff
@@ -116,7 +116,7 @@ class Event<void(ArgTs...)> {
 
     /** Configure the delay of an event
      *
-     *  @param d    Millisecond delay before dispatching the event
+     *  @param d    Chrono duration delay before dispatching the event. For example delay(5s)
```

Comment:
The original doc isn't wrong, because this `duration` is a Chrono milliseconds.

    using duration = std::chrono::duration<int, std::milli>;    // line 49

Any Chrono duration always has to be some specific resolution. This is taking the same `int` milliseconds it always did, it's just that resolution is now compiler-visible and type-checked. (And permits auto conversion so you can say `period(5s)` - that will pass 5000 milliseconds).

It will also block you from saying `delay(50us)` as that can't be losslessly converted to milliseconds.

Reply:
Good point.

Reply:
@kjbracey-arm @evedon how would you suggest we clarify the comment as it was obviously causing confusion with users? Especially as we have no documentation anywhere about how we are using Chrono times. We should be making things easy for our users not making them have to run around the houses trying to read up about how Chrono works etc!

Reply:
I'd say it makes sense to have a usage example, and the one in the "deprecated" message for the other function is not sufficient, as that in itself is going to go at some point. 

So I don't actually object to this proposed change at all. It's probably clearer, although I think I'd still want to keep the word "millisecond" in there. "Chrono millisecond delay"?

I think I was only really responding to the description and message for the PR suggesting that the original doc was wrong...

Reply:
How about something like: 
delay in milliseconds expressed as a Chrono duration. E.g. delay(5ms) ?

Reply:
Sure.

I quite liked the `5s` as it illustrated the auto-conversion, but maybe a `@param` isn't the place for that. Save it for fuller examples. There aren't many full examples in the doxygen - the only ones that spring to mind is the `ConditionVariable` methods, and those in the deprecated messages...

Reply:
Mind you I find the actual concept of this `Event` delay even more confusing. Never seen it used.

It means you've got this `Event` object you can hold or pass to someone as a `Callback`, and then when you do `cb()` on it, the event gets queued how ever many milliseconds later.

And this is useful for...?

Reply:
I don't have a strong opinion on this and I think that `delay in milliseconds expressed as a Chrono duration` is a good proposal. As for the illustration, I don't think it should go in the `@param`; since you will add more usage scenarios in the documentation, I'd suggest that you add a code snippet there.

Reply:
Agreed it's good to make it explicit that this is a chrono duration. A lot of folks won't be familiar with this concept as it's a relatively new C++ feature. @adbridge I like the "delay in milliseconds expressed as a Chrono duration. E.g. delay(5ms)" proposal.